platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +126,140 @@ public Properties loadProperties(String propertiesFile) {
-            LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
-          }
-        }
-      }
+    if (propertiesFile == null) {
+      LOGGER.debug("Properties file must not be null.");
+      return properties;
+    }
 
-      if (error || properties.isEmpty()) {
-        LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
-        File propFile = new File(propertiesFile);
-        // If properties file has fully-qualified absolute path (which
-        // the blueprint file specifies) then can load it directly.
-        if (propFile.isAbsolute()) {
-          LOGGER.debug("propertiesFile {} is absolute", propertiesFile);
-          propFile = new File(propertiesFile);
-        } else {
-          String rootDirectory = System.getProperty("karaf.home");
-          if (rootDirectory != null && !rootDirectory.isEmpty()) {
-            propFile = new File(rootDirectory, propertiesFile);
-          } else {
-            rootDirectory = System.getProperty("ddf.home");
-            if (rootDirectory != null && !rootDirectory.isEmpty()) {
-              propFile = new File(rootDirectory, propertiesFile);
-            } else {
-              propFile = new File(propertiesFile);
-            }
-          }
-        }
-        properties = new Properties();
-
-        try (InputStreamReader reader =
-            new InputStreamReader(new FileInputStream(propFile), StandardCharsets.UTF_8)) {
-          properties.load(reader);
-        } catch (FileNotFoundException e) {
-          error = true;
-          LOGGER.debug("Could not find properties file: {}", propFile.getAbsolutePath(), e);
-        } catch (IOException e) {
-          error = true;
-          LOGGER.debug("Error reading properties file: {}", propFile.getAbsolutePath(), e);
-        }
-      }
-      if (error || properties.isEmpty()) {
-        LOGGER.debug("Attempting to load properties as a resource: {}", propertiesFile);
-        InputStream ins = PropertiesLoader.class.getResourceAsStream(propertiesFile);
-        if (ins != null) {
-          try {
-            properties.load(ins);
-            ins.close();
-          } catch (IOException e) {
-            LOGGER.debug("Unable to load properties: {}", propertiesFile, e);
-          } finally {
-            IOUtils.closeQuietly(ins);
-          }
-        }
-      }
+    Iterator<BiFunction<String, ClassLoader, Properties>> strategiesIterator =
+        PROPERTY_LOADING_STRATEGIES.iterator();
+    do {
+      properties = strategiesIterator.next().apply(propertiesFile, classLoader);
+    } while (properties.isEmpty() && strategiesIterator.hasNext());
+
+    properties = substituteSystemPropertyPlaceholders(properties);
+    return properties;
+  }
+
+  /** Default property loading strategy. */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithSpring(String propertiesFile, ClassLoader classLoader) {
```

Comment:
✏️ This method can simply return `attemptLoadWithSpringAndClassLoader(propertiesFile, null)`. You'd want to update the log message in that method then to indicate whether the CL passed in was null or not.

Reply:
The logic would be slightly different in your solution.
When `attemptLoadWithSpring` is passed `null`, down the line the class loader used is `ClassUtils.getDefaultClassLoader()`.
When `attemptLoadWithSpringAndClassLoader` is passed `null`, down the line the class loader used is `PropertiesLoader.class.getClassLoader()`.